packages/uploadthing/src/client.ts

```diff
@@ -253,11 +261,12 @@ export const DANGEROUS__uploadFiles = async <TRouter extends FileRouter>(
     // Generate a URL for the uploaded image since AWS won't give me one
     const genUrl = "https://utfs.io/f/" + encodeURIComponent(fields.key);
 
-    // Poll for file data, this way we know that the client-side onUploadComplete callback will be called after the server-side version
-    await pollForFileData(presigned.key);
+    const serverData = await fetch("/api/uploadthing", {
+      headers: { "x-uploadthing-polling-key": fields.key },
+    }).then((res) => res.json());
```

Comment:
this should maybe be done in a different way, streaming / polling maybe to prevent request timeout?

Reply:
This almost feels like it should run in it's own background process if that makes sense? These behaviors are getting complex enough to need a proper state machine and some useSyncExternalStore bs

This is probably fine for now